crates/jstzd/tests/octez_client_test.rs

```diff
@@ -138,3 +144,77 @@ async fn imports_secret_key_throws() {
         res.is_err_and(|e| { e.to_string().contains("\"import\" \"secret\" \"key\"") })
     );
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn activate_protocol() {
+    // 1. start octez node
+    let (mut octez_node, _temp_data_dir) = spawn_octez_node().await;
+    // 2. setup octez client
+    let temp_dir = TempDir::new().unwrap();
+    let base_dir = temp_dir.path().to_path_buf();
+    let rpc_endpoint = Uri::from_str(octez_node.rpc_endpoint()).unwrap();
+    let rpc_endpoint: Endpoint = Endpoint::try_from(rpc_endpoint).unwrap();
+    let octez_client = OctezClientBuilder::new()
+        .set_endpoint(rpc_endpoint.clone())
+        .set_base_dir(base_dir.clone())
+        .build()
+        .unwrap();
+    // 3. import activator key
+    let activator = "activator".to_string();
+    octez_client
+        .import_secret_key(&activator, SECRET_KEY)
+        .await
+        .expect("Failed to generate activator key");
+    let params_file =
+        Path::new(std::env!("CARGO_MANIFEST_DIR")).join("tests/sandbox-params.json");
+    let blocks_head_endpoint =
+        format!("{}/chains/main/blocks/head", rpc_endpoint.to_string());
+    let response = get_response_text(&blocks_head_endpoint).await;
+    assert!(response.contains(
+        "\"protocol\":\"PrihK96nBAFSxVL1GLJTVhu9YnzkMFiBeuJRPA8NwuZVZCE1L6i\""
+    ));
+    assert!(response.contains("\"level\":0"));
+    // 4. activate the alpha protocol
+    let protocol_activated = octez_client
+        .activate_protocol(
+            "ProtoALphaALphaALphaALphaALphaALphaALphaALphaDdp3zK",
+            "0",
+            &activator,
+            &params_file,
+        )
+        .await;
+    assert!(protocol_activated.is_ok());
+    // 5. check if the protocol is activated and the block is baked.
+    let response = get_response_text(&blocks_head_endpoint).await;
+    assert!(response.contains(
```

Comment:
I think it's better to look for `"next_protocol":"ProtoALphaALphaALphaALphaALphaALphaALphaALphaDdp3zK"` because you still get `"protocol":"ProtoGenesisGenesisGenesisGenesisGenesisGenesk612im"` when alpha is actually not activated but block 1 is baked.